packages/ui/src/machines/authenticator/__tests__/defaultService.test.ts

```diff
@@ -98,6 +98,23 @@ describe('validateFormPassword', () => {
     });
   });
 
+  it.only('validates each of special characters allowed by Cognito as expected', async () => {
+    // adapted from https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-policies.html
+    const specialChars =
+      '^ $ * . [ ] { } ( ) ? " ! @ # % & / \\ , > < \' : ; | _ ~ ` = + -';
+    const specialCharsArray = specialChars.split(' ');
+    specialCharsArray.push(' '); // space is a valid character too (see docs above)
```

Comment:
Thanks for calling this out!

Reply:
Almost better than https://github.com/aws-amplify/amplify-ui/blob/4bf5e3f5ec4eba6b3405e3c17970c60d00aeaedd/packages/ui/src/machines/authenticator/__tests__/defaultService.test.ts#L80 😉